### PR TITLE
Extract an IconProps type

### PR DESCRIPTION
**Pull request by dylanbyars**
### Description
A lot of the icon components shared a collection of types. I extracted those shared types into a single type that the components could share. 

### Comments
`ArrowLineIcon`, `HouseIcon`, and `InfoIcon` buck the trend of conformity so I `Omit` a prop from those. I'm wondering if people have opinions about this. 

At first, I started thinking that the `IconProps` that I'd pulled out weren't universal _enough_ to warrant getting extracted. Upon further reflection, it feels like calling out the strangeness of the previously mentioned icon components by tying them to the shared `IconProps` type makes their novelty more obvious.

### CODE REVIEW

File: components/atoms/icons/ArrowLineIcon/ArrowLineIcon.story.jsx [hunk 1]
```diff
@@ -15,7 +15,6 @@ stories.addParameters({
 const defaultProps = () => ({
   fill: text('fill', '#000'),
   height: text('height', '44.5rem'),
-  width: text('width', '1.3rem'),
```
Comment (dylanbyars) [inline components/atoms/icons/ArrowLineIcon/ArrowLineIcon.story.jsx:18]: The `ArrowLineIcon` component doesn't recognize a width prop so I removed it.

File: components/atoms/icons/AddIcon/AddIcon.tsx [hunk 2]
```diff
@@ -1,19 +1,10 @@
 /** @module AddIcon */
 import React, {FC, memo} from 'react';
+import {IconProps} from '../icons';
 
-type AddIconProps = {
+type AddIconProps = IconProps & {
   /** Title for the svg. */
   title?: string;
```
Comment (RohitRajendran) [inline components/atoms/icons/AddIcon/AddIcon.tsx:7]: Should we just make `title` universal for all icons?

Comment (dylanbyars) [reply components/atoms/icons/AddIcon/AddIcon.tsx:7]: I love it. Will do

Comment (dylanbyars) [reply components/atoms/icons/AddIcon/AddIcon.tsx:7]: I'm going to update the storybook knobs of the components to account for this new prop but that'll take some time so I'm going to close this branch down and resume during our next innovation sprint.

Comment (dylanbyars) [reply components/atoms/icons/AddIcon/AddIcon.tsx:7]: I'll save that improvement for the next innovation sprint. This work seems mergeable (if y'all agree).

**Pull request merged**